index.bs

```diff
@@ -606,6 +701,39 @@ SessionResult = (StatusResult)
 
 ```
 
+<div algorithm>
+
+To <dfn lt="updating the event map">update the event map</dfn>, given a |list of
+event names|, a |list of contexts|, and a |target status|:
+
+  1. Let |events| be an empty set.
+
+    1. For each entry |event name| in the |list of event names|,
+       let |events| be the union of |events| and the result of [=trying=] to
+       [=obtain a set of events from a name=] with |event name|.
+
+    1. If the |list of contexts| is null, let |targets| be a set containing the [=global
+       event set=]. Otherwise for each entry |context id| in the |list of
+       contexts|, run the following substeps:
+
+      1. If there is no browsing context with [=browsing context id=] |context id| return
+         [=error=] with [=error code=] [=invalid argument=]
+
+      1. Let |context| be the browsing context with id |context id|. If the
+         [=browsing context event map=] does not contain an entry for |context|,
+         set the value of the entry for |context| to a new empty map.
+
+      1. Get the entry from the [=browsing context event map=] for |context|
+         and append it to |targets|.
+
+    1. For each |target| in |targets|, run the following steps:
+
+      1. For each |event| in events, run the following steps:
+
+        1. Set the value of |target| for |event| to |target status|.
```

Comment:
Does this work? It depends on what the type of *target* is (unclear) but *event* is a string so I think it doesn't work? Using https://infra.spec.whatwg.org/#map-set might make this clearer.